runtime/runtime.go

```diff
@@ -18,6 +20,10 @@ type Runtime interface {
 	CreateNamespace(string) error
 	// DeleteNamespace deletes a namespace in the runtime
 	DeleteNamespace(string) error
+	// CreateResource creates a runtime resource
+	CreateResource(*client.Resource, ...client.CreateOption) error
```

Comment:
We should not be leaking references to util. runtime should define Resource type itself. Talk to @ben-toogood about this as he has the most knowledge about what we're attempting to do here.

We have two potential avenues. Entirely rewrite the interface to only deal with Resource or add CreateResource/DeleteResource methods and separately refactor the other methods to Run/Kill/Update/Status.

Reply:
Agreed. I've included examples of how to do the namesapces using Create/DeleteResource in the unit test. I'd leave removing CreateNamespace until I have a solution that works for you though.

Reply:
The issue is specifically related to importing "github.com/micro/go-micro/v3/util/kubernetes/client" and requiring the interface to accept client.Resource. This needs to be a Resource type we define in this runtime package and map through. The fact that we have the util package is just a nice to have, the implementation could very well live in the same place but its separate because its useful for multiple components.

Reply:
OK, I've made some changes which improve this.